sdk/file_sdk/api/istream_handler.h

```diff
@@ -28,18 +28,16 @@ class IStreamHandler {
 
   virtual std::shared_ptr<Tag> GetLabel() = 0;
 
-  virtual void SetLabel(const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
+  virtual std::string SetLabel(std::shared_ptr<IStream> outputStream, const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
 
-  virtual void SetLabel(std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
+  virtual std::string SetLabel(std::shared_ptr<IStream> outputStream, std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
 
-  virtual void DeleteLabel(const std::string& justificationMessage) = 0;
+  virtual std::string DeleteLabel(std::shared_ptr<IStream> outputStream, const std::string& justificationMessage) = 0;
 
   virtual void SetProtection(const UserPolicy& policy ) = 0;
```

Comment:
change to the new class

Reply:
next PR to seprate to P and LP handler

---
In reply to: [141048268](https://github.com/AzureAD/rms-sdk-for-cpp/pull/170#discussion_r141048268) [](ancestors = 141048268)

Reply:
Let's do the move to MIP first before you start seperating it.